src/GraphRenderer.ts

```diff
@@ -1097,21 +1097,26 @@ export class GraphRenderer {
         // select handlers
         if(node !== null && event.button != 1 && !event.shiftKey){
 
+            //double click has highest priority, select only this node if it is a contruct
+            if(GraphRenderer.dragSelectionDoubleClick && node.isGroup()) {
+                eagle.setSelection(node, Eagle.FileType.Graph);
+            }   
+
+            //check for alt clicking, if so, add the target node and its children to the selection
+            else if(event.button != 2 && !event.altKey&&node.isGroup() && !eagle.objectIsSelected(node)){
+                GraphRenderer.selectNodeAndChildren(node,GraphRenderer.shiftSelect)
+            }
+
             // check if shift key is down, if so, add or remove selected node to/from current selection | keycode 2 is the middle mouse button
-            if (node !== null && event.shiftKey && !event.altKey){
+            else if (node !== null && event.shiftKey && event.altKey){
```

Comment:
I don't understand the check for altKey AND shiftKey here. Could you add a comment?

Reply:
looking at it again this part did nothing, there was a duplicate in the code. shift click on a construct adds or removes the construct and its children from the selection. shift + alt click on a construct adds or removes just the construct, without children.